torch/nn/modules/transformer.py

```diff
@@ -196,27 +196,28 @@ def __init__(self, encoder_layer, num_layers, norm=None, enable_nested_tensor=Tr
         self.use_nested_tensor = enable_nested_tensor
         self.mask_check = mask_check
 
-        first_layer = self.layers[0]
-        str_first_layer = "self.layers[0]"
-        why_not_sparsity_fast_path = ''
-        if not isinstance(first_layer, torch.nn.TransformerEncoderLayer):
-            why_not_sparsity_fast_path = f"{str_first_layer} was not TransformerEncoderLayer"
-        elif first_layer.norm_first :
-            why_not_sparsity_fast_path = f"{str_first_layer}.norm_first was True"
-        elif not first_layer.self_attn.batch_first:
-            why_not_sparsity_fast_path = f" {str_first_layer}.self_attn.batch_first was not True"
-        elif not first_layer.self_attn._qkv_same_embed_dim:
-            why_not_sparsity_fast_path = f"{str_first_layer}.self_attn._qkv_same_embed_dim was not True"
-        elif not first_layer.activation_relu_or_gelu:
-            why_not_sparsity_fast_path = f" {str_first_layer}.activation_relu_or_gelu was not True"
-        elif not (first_layer.norm1.eps == first_layer.norm2.eps) :
-            why_not_sparsity_fast_path = f"{str_first_layer}.norm1.eps was not equal to {str_first_layer}.norm2.eps"
-        elif first_layer.self_attn.num_heads % 2 == 1:
-            why_not_sparsity_fast_path = "num_head is odd"
-
-        if enable_nested_tensor and why_not_sparsity_fast_path:
-            warnings.warn(f"enable_nested_tensor is True, but self.use_nested_tensor is False because {why_not_sparsity_fast_path}")
-            self.use_nested_tensor = False
+        if self.num_layers > 0:
```

Comment:
replace all references to `first_layer` in `__init__` with references to `encoder_layer` and make num_layers another reason why to set `use_nested_tensor=False`